corehq/ex-submodules/casexml/apps/case/mock/case_block.py

```diff
@@ -2,145 +2,167 @@
 from datetime import datetime, date
 import numbers
 from xml.etree import ElementTree
-from casexml.apps.case.xml import NS_VERSION_MAP, V2
+from casexml.apps.case.xml import V2_NAMESPACE
 from dimagi.utils.parsing import json_format_datetime
+from collections import namedtuple
+from functools import partial
 
+# relationship = "child" for index to a parent case (default)
+# relationship = "extension" for index to a host case
+IndexAttrs = namedtuple('IndexAttrs', ['case_type', 'case_id', 'relationship'])
+ChildIndexAttrs = partial(IndexAttrs, relationship='child')
 
-class CaseBlock(dict):
+
+class CaseBlock(object):
+    """
+    see https://github.com/dimagi/commcare/wiki/casexml20 for spec
+    """
     undefined = object()
 
     def __init__(self, case_id, date_modified=None, user_id=undefined,
                  owner_id=undefined, external_id=undefined, case_type=undefined,
                  case_name=undefined, create=False, date_opened=undefined, update=None,
-                 close=False, index=None, strict=True):
-        """
-        see https://github.com/dimagi/commcare/wiki/casexml20 for spec
-        """
-        super(CaseBlock, self).__init__()
-        self._id = case_id
+                 close=False, index=None):
+        # todo: can we use None instead of CaseBlock.undefined, throughout?
+        owner_id = CaseBlock.undefined if owner_id is None else owner_id
+        self.update = copy.copy(update) if update else {}
         now = datetime.utcnow()
-        date_modified = date_modified or now
-        if date_opened == CaseBlock.undefined and create:
-            date_opened = now.date()
-        update = copy.copy(update) if update else {}
-        index = copy.copy(index) if index else {}
-
-        self.XMLNS = NS_VERSION_MAP.get(V2)
-
-        self.VERSION = V2
-        self.CASE_TYPE = "case_type"
-
-        if create:
-            self['create'] = {}
-            # make case_type
-            case_type = "" if case_type is CaseBlock.undefined else case_type
-            case_name = "" if case_name is CaseBlock.undefined else case_name
-            owner_id = "" if owner_id is CaseBlock.undefined else owner_id
-        self['update'] = update
-
-        create_or_update = {
-            self.CASE_TYPE: case_type,
-            'case_name': case_name,
-        }
+        self.date_modified = date_modified or now
+        self.date_opened = (now.date() if create and date_opened is CaseBlock.undefined
+                            else date_opened)
+
+        self.case_type = "" if create and case_type is CaseBlock.undefined else case_type
```

Comment:
Realizing you didn't change this but should case type not be compulsory for create?